editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java

```diff
@@ -299,37 +307,67 @@ private void generate(IProgressMonitor monitor) throws Exception {
-				initContextForTool(isBlueprint, isSpringBoot, camelFile, tool);
-				if (!Strings.isEmpty(options.getTargetServiceAddress())) {
-					URI targetAddressURI = new URI(options.getTargetServiceAddress());
-					URL targetAddressURL = targetAddressURI.toURL();
-					tool.setJaxwsAddress(targetAddressURL);
-				}
-				if (!Strings.isEmpty(options.getTargetRestServiceAddress())) {
-					URL targetRestAddressURL = new URL(options.getTargetRestServiceAddress());
-					tool.setJaxrsAddress(targetRestAddressURL);
-				}
-				if (outJavaPath != null && outJavaPath.toFile().exists()) {
-					tool.setJavaOut(outJavaPath);
-				}
-				ClassLoader loader = tool.getClass().getClassLoader();
-				Thread.currentThread().setContextClassLoader(loader);
-				tool.process();
-				updateDependencies(subMon.split(1));
-			}  finally {
-				Thread.currentThread().setContextClassLoader(oldLoader);
-				project.refreshLocal(IResource.DEPTH_INFINITE, subMon.split(1));
-			}
+			doWsdl2RestMigration(project, javaFile, wsdlLocation, isBlueprint, isSpringBoot, camelFile, subMon.split(2));
 		}
 		subMon.setWorkRemaining(0);
+		done = true;
 	}
 
+	private void doWsdl2RestMigration(IProject project, File javaFile, URL wsdlLocation, boolean isBlueprint, boolean isSpringBoot, File camelFile, IProgressMonitor monitor) throws IOException, CoreException, URISyntaxException  {
+		SubMonitor subMon = SubMonitor.convert(monitor, 2);
+		Path contextPath;
+		ClassLoader oldLoader = Thread.currentThread().getContextClassLoader();
+		try {
+			Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+
+			Path outJavaPath = javaFile.toPath();
+			Wsdl2Rest tool = new Wsdl2Rest(wsdlLocation, outJavaPath);
+			contextPath = initContextForTool(isBlueprint, isSpringBoot, camelFile, tool);
+			if (!Strings.isEmpty(options.getTargetServiceAddress())) {
+				URI targetAddressURI = new URI(options.getTargetServiceAddress());
+				URL targetAddressURL = targetAddressURI.toURL();
+				tool.setJaxwsAddress(targetAddressURL);
+			}
+			if (!Strings.isEmpty(options.getTargetRestServiceAddress())) {
+				URL targetRestAddressURL = new URL(options.getTargetRestServiceAddress());
+				tool.setJaxrsAddress(targetRestAddressURL);
+			}
+			if (outJavaPath != null && outJavaPath.toFile().exists()) {
+				tool.setJavaOut(outJavaPath);
+			}
+			ClassLoader loader = tool.getClass().getClassLoader();
+			Thread.currentThread().setContextClassLoader(loader);
+			try {
+				tool.process();	
+			} catch (Exception ex) {
+				throw new IOException(ex);
+			}				
+			updateDependencies(subMon.split(1));
+		}  finally {
+			Thread.currentThread().setContextClassLoader(oldLoader);
+			project.refreshLocal(IResource.DEPTH_INFINITE, subMon.split(1));
+		}
+		if (contextPath != null) {
+			Job job = new Job("Update Camel Context File...") {
+				
+				@Override
+				protected IStatus run(IProgressMonitor monitor) {
+					SubMonitor subMon = SubMonitor.convert(monitor, 1);
+					while (!done) {
```

Comment:
I fear that if an exception is throw the done which is not in a finally block will stay at false and this job will loop forever